src/lib/listeners/LogEventListener.ts

```diff
@@ -0,0 +1,65 @@
+import {CodeAnalyzer, EngineLogEvent, EventType, LogEvent, LogLevel} from '@salesforce/code-analyzer-core';
+import {Display} from '../Display';
+import {LogWriter} from '../writers/LogWriter';
+
+export interface LogEventListener {
+	listen(codeAnalyzer: CodeAnalyzer): void;
+}
+
+export class LogEventDisplayer implements LogEventListener {
+	private readonly display: Display;
+
+	public constructor(display: Display) {
+		this.display = display;
+	}
+
+	public listen(codeAnalyzer: CodeAnalyzer): void {
+		// Set up listeners
+		codeAnalyzer.onEvent(EventType.LogEvent, (e: LogEvent) => this.handleEvent('Core', e));
+		codeAnalyzer.onEvent(EventType.EngineLogEvent, (e: EngineLogEvent) => this.handleEvent(e.engineName, e));
+	}
+
+	private handleEvent(source: string, event: LogEvent|EngineLogEvent): void {
+		// We've arbitrarily decided to log only events of type "Info" or higher, to avoid potentially flooding the CLI
+		// with a ton of noisy statements from other engines. At some point in the future, we may make this configurable.
+		if (event.logLevel > LogLevel.Info) {
+			return;
+		}
+		const formattedMessage = `${source} [${formatTimestamp(event.timestamp)}]: ${event.message}`;
+		switch (event.logLevel) {
+			// We display errors at the "warning" level, because calling `.error()` would actually kill the transaction,
+			// and we probably don't want that.
+			case LogLevel.Error:
```

Comment:
Note that there are error level logs from the core side. And they occur prior to the run method... specifically one of them happens if you try to add in a plugin but the engine can't be created because it doesn't validate the requirements. For example, if PMD doesn't have the right java version. 

I left this as an error log event so that you can decide to either throw an error on the client... or skip that engine and continue on with a warning message.

Reply:
So just to be sure I understand, there are cases where Core will log an `error`-level event (and presumably perform a no-op) instead of throwing an exception?